### DISCUSSION TITLE
Aggiornamento Firmware Sonoff Zigbee Dongle-E

### DISCUSSION THREAD

**Discussion by Dekard78** [indomus/forum#1974] [Hardware & Firmware]
Ciao a tutti

spero possiate darmi una mano. Faccio una doverosa premessa: sono nuovo (molto nuovo...) del mondo Home Assistant ect... perciò, può essere che la domanda che vado a porre sia abbastanza stupida. 
Ho allestito un pc di recupero nel quale ho installato (con molta difficoltà Home Assitant verisone base) e preso la chiavetta in oggetto. Adesso avrei la necessità di aggiornare il firmware, per poterla installare in HA. E qui mi son bloccato, ho scaricato il firmware (seguendo guida), ma il dongle non va in bsl in nessuna maniera: quando parte l'eseguibile per l'aggiornamento mi ridà questo errore

Timeout waiting for ACK/NACK after 'Synch (0x55 0x55)'

Faccio presente che le ho provate tutte: pulsante Boot per svariati secondi, pulsante Boot + Reset ect.... ma l'errore rimane e chiaramente il firmware si blocca.
Qualcuno ha qualche idea? Potrei decidere di buttare tutto dalla finestra, ma mi pare un peccato... 

Grazie per le vostre risposte....

**Comment by snakuzzo** [comment]
Ciao.
Premesso che dovresti aver ricevuto la chiavetta già con il firmware pre-caricato, leggi l' [ARTICOLO DI OGGI](https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zigbee-3-0-usb-dongle-plus-v2-zbdongle-e/)

**Comment by Dekard78** [comment]
Ciao, ti ringrazio per il link, ma purtroppo non cambia niente. Ho seguito quanto scritto nella guida: il dongle viene visto come 

COM3 USB-ENHANCED-SERIAL CH9102

Quando provo a inviare il firmware alla chiavetta mi restituisce l'errore

Transferring , errors 1... Press Ctrl+C to cancel.

Il fatto del firmware, potrebbe anche essere non rilevante se HA vedesse la chiavetta. Ma non ne vuole assolutamente sapere di vederla come antenna. Tra l'altro lo stesso problema lo riscontro in Windows (Ver.PRO1 10). La chiavetta viene vista come USB generico....

Non esiste un altro metodo per farla riconoscere ad HA?

Grazie in anticipo

**Comment by Dekard78** [comment]
Ciao ti ringrazio, e scusa il ritardo nella risposta. Ho letto e provato a seguire passo passo la guida ma senza successo (ho risposto anche nel forum). Il problema è che non riesco a flashare il firmware nella chiavetta (che viene vista come generica da WIN10 Pro, e non viene proprio vista da HA). Immagino sia io a sbagliare (la chiavetta non va in Bls)…Ma non c’è un altro modo per farla integrare in Ha?Ti ringrazio ancora x avermi scritto.Il giorno 10 ott 2022, alle ore 11:36, Diego Ambrosanio ***@***.***> ha scritto:﻿
Ciao.
Leggi l' ARTICOLO DI OGGI

**Reply by snakuzzo** [reply]
ma tu sei sicuro di aver comprato la chiavetta giusta ?

**Reply by snakuzzo** [reply]
...e ti prego non fare reply dalla mail...grazie

**Comment by Dekard78** [comment]
Scusate non sapevo che rispondendo alla mail, si generasse un reply direttamente sul forum. La chiavetta è corretta:

SONOFF ZBDONGLE-E USB Dongle Plus.

Ho provato anche su di un altro pc, più recente (sempre con Windows 10 Pro) viene riconosciuta come "Dispositivo Seriale USB (Com6)".

**Reply by KC-inDomus** [reply]
perché evidentemente ha bisogno di un driver

**Reply by snakuzzo** [reply]
concordo

**Reply by snakuzzo** [reply]
ma aspetta...stai seguendo la guida corretta @Dekard78 ?

https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zigbee-3-0-usb-dongle-plus-v2-zbdongle-e/

o stai seguendo quella per la versione "P" ?

https://indomus.it/guide/riprogrammare-il-firmware-di-itead-sonoff-zigbee-3-0-usb-dongle-plus/

**Comment by Dekard78** [comment]
Sto seguendo la guida corretta: Versione E, e concordo assolutamente con voi che ho bisogno di un driver da dare a Windows perché venga rilevata. Il problema è che non ho trovato nulla a riguardo (probabilmente o ho cercato male io o proprio non esiste). 
Il fatto comunque rimane: installato Hassio senza problemi, rimane solo l’antenna. Ho visto che YouTube che un user (inglese) che ho postato un videro abbastanza semplice per l’integrazione del domgle-e. Proverò a seguirlo…

**Reply by snakuzzo** [reply]
Trovato...
I driver per windows li trovi qui...

https://sonoff.tech/product-review/sonoff-zigbee-3-0-usb-dongle-plus-tutorials/#2

**Comment by Dekard78** [comment]
Vi ringrazio tutti, per il tempo che avete speso, e mi scuso se, in qualche modo le domande sono state perlopiù stupide. Ho scaricato i driver, installati, ma non è cambiato niente. La chiavetta non viene vista, se non come CH9102 e non ne vuole sapere di farsi aggiornare (vedi allegato per vedere cosa mostra windows). 

A questo punto, dopo averle provate tutte, penso che la chiavetta farà un volo nel pattume. Vi chiedo un ultima cortesia: la versione precedente (la P per intenderci) è pronta all'uso su HA o dovrei fare tutta la trafila del firmware? Perchè se è un "pronti via" posso anche pensare di prenderne un altra, altrimenti penso proprio che il progetto HA andrà nel dimenticatoio....

Grazie a tutti per risposte.

https://i.postimg.cc/kGFZ4Q1f/Nuova-immagine-bitmap.png

**Reply by snakuzzo** [reply]
Esagerato.
Vedremo di riprovare il giro anche su Windows, ma devi considerare che la chiavetta arriva già preflashata (che sia modello P o E)
Ma mi chiedo...e ti chiedo...hai visto "bene" il link che ti ho passato? Se scorri in basso è indicato un altro metodo su come flashare...sicuramente più macchinoso, ma almeno provalo.

**Reply by snakuzzo** [reply]
E comunque...prima di "buttare tutto" prova con una VM Linux, un rpi, quello che è.
Parere personale...sbagli ad avere un approccio di questo tipo. Come ti dicevo prima è esagerato buttare tutto per non riuscire a flashare una chiavetta zigbee

**Reply by KC-inDomus** [reply]
> Vi ringrazio tutti, per il tempo che avete speso, e mi scuso se, in qualche modo le domande sono state perlopiù stupide. Ho scaricato i driver, installati, ma non è cambiato niente. La chiavetta non viene vista, se non come CH9102 e non ne vuole sapere di farsi aggiornare (vedi allegato per vedere cosa mostra windows).
> 
> A questo punto, dopo averle provate tutte, penso che la chiavetta farà un volo nel pattume. Vi chiedo un ultima cortesia: la versione precedente (la P per intenderci) è pronta all'uso su HA o dovrei fare tutta la trafila del firmware? Perchè se è un "pronti via" posso anche pensare di prenderne un altra, altrimenti penso proprio che il progetto HA andrà nel dimenticatoio....
> 
> Grazie a tutti per risposte.
> 
> https://i.postimg.cc/kGFZ4Q1f/Nuova-immagine-bitmap.png

credo che saranno più gli utenti che ragionano così, a finire nel dimenticatoio...

**Comment by Dekard78** [comment]
Ciao, ordinata versione P. Domani arriva e vediamo se cambia qualcosa. Nel caso di nulla di fatto, step successivo è cambiare il PC. A mio avviso, pur essendo particolarmente performante per quello che deve fare, è troppo anziano. La MB è del 2013 e ho fatto fatica ad aggiornare il bios anche solo per installare Docker (al primo tentativo di installare HA). Per quanto riguarda il “mollare”, a mio avviso, bisogna anche capire che in molti non hanno le conoscenze di base per affrontare un installazione di questo genere. Se ci si intoppa in qualche problema che esula dalle varie guide, nel 90% dei casi si va in pallone e non se ne esce più (un po’ come sta succedendo a me).

Seguono aggiornamenti, e comunque grazie ancora a tutti!

**Reply by snakuzzo** [reply]
Sì ma ti stiamo dicendo che stai sbagliando completamente approccio. Non va la chiavetta e cambi PC? 😅😅😅 Non è assurdo...di più.

Sul discorso delle competenze, quelle vengono col tempo...ma di sicuro non si ottengono con il buttare tutto o cambiare PC per cose così...dai.
Se vai a cannone in questo modo, ti dico da subito che a sto punto è meglio lasciar perdere tutto quanto

**Comment by Dekard78** [comment]
L’ho detto fin da subito. Il fatto che il dongle “non funziona”, molto probabilmente è causato dall’utilizzatore (io nello specifico). Discorso diverso diverso per il pc: prima di acquistare un nuc/minipc/rpi ho fatto questo tentativo con quello che avevo in casa (un mini desktop Lenovo del 2013) nel caso fosse andato a buon fine (cosa che spero tutt’ora) avrei fatto l’acquisto. Ripeto: non voglio imputare tutte le colpe alla parte hardware, ma onestamente, più di un dubbio mi è venuto (e continuo ad averlo). Domani vi aggiorno, anche perché è diventata una questione d’orgoglio 😵‍💫😭

**Reply by snakuzzo** [reply]
Ok...fatto bene prima a provare.
Se non riesci proprio a risolvere e se puoi e sei pratico, crea una VM Debian ad esempio...e prova.

**Comment by Anacidol** [comment]
Salve a tutti.
Sto tentando di installare il firmware come router, però mi blocco all'istruzione `python3 -m pip3 install -r requirements.txt` , La risposta che il terminal mi da è `/usr/bin/python3: No module named pip3` dove sto sbagliando?

**Reply by snakuzzo** [reply]
Ciao.
Le cose sono due...

O non hai il symlink `pip3` ... Prova con questo e vedi se va...

```
which pip
```

e se te lo trova...

```
python3 -m pip install -r requirements.txt
```

Oppure non hai proprio installato `pip` . In tal caso

```
sudo apt-get install python3-pip
```

e poi riprovi

**Reply by Anacidol** [reply]
Grazie Snakuzzo per la risposta.

`which pip` mi da come risposta `bash: wich: comando non trovato`.
Ho provato `pip --version` e la risposta è `pip 22.2.2 from /home/pi/.local/lib/python3.7/site-packages/pip (python 3.7)`, quindi penso che pip sia installato.
Per sicurezza ho digitato `sudo apt-get install python3-pip`, la risposta è stata `python3-pip è già alla versione più recente (18.1-5+rpt1).` quindi purtroppo sono sempre bloccato.

Che il problema sia perché sto cercando di fare questo flash con un raspberry con Raspian 10? magari non è adatto...

**Reply by snakuzzo** [reply]
Hai sbagliato a scrivere `which`.
Ritenta :)

**Reply by Anacidol** [reply]
che sbadato!, scusami... ritento :)
Al comando `which` risponde `/home/pi/.local/bin/pip`
quindi ho riprovato a dare questo comando: `python3 -m pip install -r requirements.txt`. Mi da questo messaggio:

> Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: pyserial==3.4 in /usr/lib/python3/dist-packages (from -r requirements.txt (line 1)) (3.4)
Requirement already satisfied: xmodem==0.4.6 in /home/pi/.local/lib/python3.7/site-packages (from -r requirements.txt (line 2)) (0.4.6)

**Reply by snakuzzo** [reply]
Non hai errori...quindi mi sembra tutto ok

**Reply by Anacidol** [reply]
ok, allora procedo secondo la guida, però all'ultimo comando: 

```
python3 Elelabs_EzspFwUtility.py flash -p /dev/ttyACM0 -f ./Z3RouterUSBDonlge_EZNet6.10.3_V1.0.0.gbl
```
ottengo questo:
```
2022/10/15 20:24:36 Elelabs_EzspFwUtility:   EZSP adapter in bootloader mode detected:
2022/10/15 20:24:36 Elelabs_EzspFwUtility:   Gecko Bootloader v1.12.00
2022/10/15 20:24:36 Elelabs_EzspFwUtility:   Allready in bootloader mode. No need to restart
2022/10/15 20:24:37 Elelabs_EzspFwUtility:   Successfully restarted into X-MODEM mode! Starting upload of the new firmware... DO NOT INTERRUPT(!)
send error: expected ACK; got b'\x18' for block 1
send error: expected ACK; got b'\x18' for block 1
send error: expected ACK; got b'\x18' for block 1
send error: expected ACK; got b'\r' for block 1
send error: expected ACK; got b'\n' for block 1
send error: expected ACK; got b'S' for block 1
send error: expected ACK; got b'e' for block 1
send error: expected ACK; got b'r' for block 1
send error: expected ACK; got b'i' for block 1
send error: expected ACK; got b'a' for block 1
send error: expected ACK; got b'l' for block 1
send error: expected ACK; got b' ' for block 1
send error: expected ACK; got b'u' for block 1
send error: expected ACK; got b'p' for block 1
send error: expected ACK; got b'l' for block 1
send error: expected ACK; got b'o' for block 1
send error: expected ACK; got b'a' for block 1
send error: NAK received 17 times, aborting.

2022/10/15 20:24:44 Elelabs_EzspFwUtility:   Firmware upload failed. Please try a correct firmware image or restart in normal mode.
```
sembra che vada a formare "Serial uploa", strano...

**Reply by snakuzzo** [reply]
Non saprei...
Curiosità...ti dà stesso problema con il firmware coordinator?

**Reply by Anacidol** [reply]
Risolto!, tutta colpa mia... :(

Ho provato con il firmware coordinator come da te suggerito e funziona. 
Però c'è da dire ho copia incollato il link della guida, quindi per forza corretto.
Allora mi è venuto il dubbio che il link che ho ricavato da github per il router fosse errato. L'ho scaricato sul pc e ho notato che la dimensione era diversa da quella sul raspi (#°ç&%$"£# <-parolaccia).
Quindi, errore mio.
il commando corretto per il firmware router era:

```
wget https://github.com/itead/Sonoff_Zigbee_Dongle_Firmware/raw/master/Dongle-E/Router/Z3RouterUSBDonlge_EZNet6.10.3_V1.0.0.gbl
```
Posso sugerire di riportarlo sulla guida? così si evitano guai...

Comunque grazie per l'assistenza!

**Reply by snakuzzo** [reply]
Ok, ma non c'è nulla da correggere nella guida. Noi abbiamo fatto l'esempio con il file del firmware coordinator. Viene da sé che devi fare lo stesso giro per scaricare quello di tipo router.

**Reply by KC-inDomus** [reply]
(e c'è scritto)

**Comment by Dekard78** [comment]
Aggiornamento: arrivata versione P. Anche in questo caso, tutto fermo. Non riconosciuta e quindi sono punto a capo. Provo a reinstallare da capo Windows (per evitare di aver installato erroneamente qualche driver che interferisce) e vi aggiorno. Magari, se riesco, seguo il vostro suggerimento e metto su una partizione con Linux, nella speranza che cambi qualcosa....

**Comment by Dekard78** [comment]
Aggiornamento: chi la dura la vince. Installato da zero Ubuntu. Installato HASV, inserito il primo dongle (la verisone E per capirci) riconosciuta al primo colpo senza fare nessun tipo di aggiornamento, come avete detto voi (il firmware era già all'interno). Quindi adesso ho un dongle (la versione P) in più (magari la rendo ad Amazon...). 

Ho un piccolo intoppo nuovo, però (e te pareva...). Quando vado per cercare di installare Mosquito, mi esce questo, e non mi fa andare avanti:

" 'AddonManager.install' blocked from execution, system is not healthy"

Da quello che ho letto in giro, è un messaggio che può dire tutto e niente... Avete qualche info in più?

Grazie ancora a tutti!

**Reply by snakuzzo** [reply]
Se per caso hai portainer...rimuovilo e riprova.
